### PR TITLE
[Spec] update spec for vesting.move

### PR DESCRIPTION
**Pull request by UIZorrot**
### Description

We finished the basic properties specification of vesting.move without the vector loop
There are some works remaining.

### Test Plan
aptos move prove -f vesting

### CODE REVIEW

File: aptos-move/framework/aptos-framework/boogie.bpl
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/boogie.bpl:1]: please remove this file

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move [hunk 1]
```diff
@@ -45,8 +45,16 @@ spec aptos_framework::vesting {
...
-        pragma verify = false;
+        pragma verify_duration_estimate = 120;
+
+        include TotalAccumulatedRewardsAbortsIf;
+    }
+
+    spec schema TotalAccumulatedRewardsAbortsIf {
+        vesting_contract_address: address;
+
+        requires staking_contract.commission_percentage >= 0 && staking_contract.commission_percentage <= 100;
```
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/sources/vesting.spec.move:56]: Could you briefly explain why we need this requirement here?

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move [hunk 2]
```diff
@@ -70,19 +78,47 @@ spec aptos_framework::vesting {
         aborts_if total_active_stake < staking_contract.principal;
         aborts_if accumulated_rewards * staking_contract.commission_percentage > MAX_U64;
         aborts_if (vesting_contract.remaining_grant + commission_amount) > total_active_stake;
+        aborts_if total_active_stake < vesting_contract.remaining_grant;
     }
 
     spec accumulated_rewards(vesting_contract_address: address, shareholder_or_beneficiary: address): u64 {
-        // TODO: Uses `total_accumulated_rewards` which is not verified.
+        // TODO: A severe timeout can not be resolved.
```
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/sources/vesting.spec.move:85]: Could you identity which properties lead to timeout?

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move [hunk 3]
```diff
@@ -99,49 +135,111 @@ spec aptos_framework::vesting {
...
+        aborts_if withdrawal_address == @aptos_framework || withdrawal_address == @vm_reserved;
+        aborts_if !exists<account::Account>(withdrawal_address);
+        aborts_if !exists<coin::CoinStore<AptosCoin>>(withdrawal_address);
+        aborts_if len(shareholders) == 0;
+        aborts_if simple_map::spec_len(buy_ins) != len(shareholders);
     }
 
     spec unlock_rewards(contract_address: address) {
         // TODO: Calls `unlock_stake` which is not verified.
+        // Current verification times out even with partial.
```
Comment (rahxephon89) [inline aptos-move/framework/aptos-framework/sources/vesting.spec.move:149]: Could you identify which aborts_if conditions lead to timeout?

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move (hunk 1 above)
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/vesting.spec.move:59]: I add comment there

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move (hunk 3 above)
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/vesting.spec.move:152]: I add comment in the spec, it's mainly from TotalAccumulatedRewardsAbortsIf

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move [hunk 4]
```diff
@@ -70,19 +81,48 @@ spec aptos_framework::vesting {
...
+        aborts_if total_active_stake < vesting_contract.remaining_grant;
     }
 
     spec accumulated_rewards(vesting_contract_address: address, shareholder_or_beneficiary: address): u64 {
-        // TODO: Uses `total_accumulated_rewards` which is not verified.
-        pragma verify = false;
+        // TODO: A severe timeout can not be resolved.
+        pragma verify_duration_estimate = 1000;
+
+        // This schema lead to timeout
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/vesting.spec.move:91]: Do we know why? Is it proved with a large time-limit?

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move [hunk 5]
```diff
@@ -99,49 +139,102 @@ spec aptos_framework::vesting {
...
         // TODO: Calls `unlock_stake` which is not verified.
+        // Current verification times out.
         pragma verify = false;
+        include UnlockRewardsAbortsIf;
+    }
+
+    spec schema UnlockRewardsAbortsIf {
+        contract_address: address;
+
+        // Cause timeout here
```
Comment (junkil-park) [inline aptos-move/framework/aptos-framework/sources/vesting.spec.move:159]: Is this the same issue as the one above? Is it proved with a enough time-limit?

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move (hunk 4 above)
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/vesting.spec.move:92]: This two item both contribute to the timeout, the first one deal much more
aborts_if (vesting_contract.remaining_grant + commission_amount) > total_active_stake;
aborts_if total_active_stake < vesting_contract.remaining_grant;

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move (hunk 5 above)
Comment (UIZorrot) [reply aptos-move/framework/aptos-framework/sources/vesting.spec.move:160]: yes, its the same issue

File: aptos-move/framework/aptos-framework/sources/vesting.spec.move (hunk 4 above)
Comment (junkil-park) [reply aptos-move/framework/aptos-framework/sources/vesting.spec.move:93]: Was it proved when you gave a large time-limit?

File: aptos-move/framework/cached-packages/generated/head.mrb
Comment (rahxephon89) [inline aptos-move/framework/cached-packages/generated/head.mrb:1]: Could you rebase and see whether we still need this file?

Comment (UIZorrot) [reply aptos-move/framework/cached-packages/generated/head.mrb:1]: it seemed in the newest aptos repo, there is no head.mrb exsit, should i just delete it?

Comment (rahxephon89) [reply aptos-move/framework/cached-packages/generated/head.mrb:1]: Yeah, the current status should be fine

**Review by rahxephon89 — APPROVED** [review]

**Review by junkil-park — APPROVED** [review]

**Pull request merged**